res/controllers/common-hid-utils.js

```diff
@@ -0,0 +1,83 @@
+HIDController.prototype.defaultScalers = {
+    'volume': function (value) { return script.absoluteLin(value, 0, 1, 0, 4096) },
+    'volumefader': function (value) { return Math.pow(HIDController.prototype.defaultScalers.volume(value), 2) },
+    'eq': function (value) { return script.absoluteLin(value, 0, 2, 0, 4096) },
+    'pregain': function (value) { return script.absoluteNonLin(value, 0, 1, 4, 0, 4096) },
+    'rate': function (value) { return script.absoluteLin(value, -1, 1, 0, 4096) },
+    'plusminus': function (value) { return script.absoluteLin(value, -1, 1, -4096, 4096) }
+}
+
+HIDController.prototype.softTakeover = function() {
+    for(var i = 1; i < arguments.length; i++) {
+        engine.softTakeover(arguments[0], arguments[i], true);
+    }
+}
+
+HIDController.prototype.softTakeoverAll = function() {
```

Comment:
No, it actually covers all controls I have had a controller control so far. That is why it is separate. As you will see, these util functions are there because they are also used in my Kontrol F1 script.

Reply:
There are far more ControlPotmeters than this in Mixxx. Just because this small list works for the two controllers you have does not mean it is generalizable to all controllers.

Reply:
That is why it is in my utils file. I am not saying that it works for everything. But feel free to expand it to include every control, I always want all controls in my controllers in softTakeover, that's why I have this. 
Maybe you could even consider an option for global softTakeover. I don't see the benefit of having the scripter decide what to softTakeover, this is much more a personal style decision.

Reply:
> Maybe you could even consider an option for global softTakeover. I don't see the benefit of having the scripter decide what to softTakeover, this is much more a personal style decision.

Good point, I think that is worth discussing further. However, I don't think putting every ControlPotmeter into a big JS function is a good way to implement that (maintaining it would be a pain). Could you start a new Zulip topic about it to see what others think?

Reply:
> I don't think putting every ControlPotmeter into a big JS function is a good way to implement that (maintaining it would be a pain). 

Yeah, but it is the only way to currently do it, because maintaining it in one file is easier than for every controller separately.

> Could you start a new Zulip topic about it to see what others think?

Sure: https://mixxx.zulipchat.com/#narrow/stream/109122-general/topic/Controller.20soft.20takeover/near/160308287